spec/integration/defaults_spec.rb

```diff
@@ -126,12 +126,13 @@
     end
 
     it "path should support UTF8 characters" do
+      case_preserved_key = ENV.keys.grep(/^PATH$/i).first
       path = "c:\\windows\\system32#{File::PATH_SEPARATOR}c:\\windows#{File::PATH_SEPARATOR}C:\\" + rune_utf8
       Puppet::Util.withenv( {"PATH" => path }, :windows) do
         Puppet.settings[:path] = "none" # this causes it to ignore the setting
 
         envhash = Puppet::Util::Windows::Process.get_environment_strings
-        expect(envhash['Path']).to eq(path)
+        expect(envhash[case_preserved_key]).to eq(path)
```

Comment:
Should this be using `Puppet::Util.get_env` ?  It already does insensitive env name lookups?

https://github.com/puppetlabs/puppet/blob/605187329a42e200d011cbccfd9e79eb4de02145/lib/puppet/util.rb#L36-L52

Side note.  Should probably use `.find` in that code instead of `.each` but much of a much-ness

Reply:
Updated to `get_env`. I kept `each` because we need the value of the key, and `find` just returns the key.